sdks/java/core/src/main/java/org/apache/beam/sdk/io/GenerateSequence.java

```diff
@@ -98,6 +103,46 @@
     abstract GenerateSequence build();
   }
 
+  /** Exposes GenerateSequence as an external transform for cross-language usage. */
+  @AutoService(ExternalTransformRegistrar.class)
+  public static class External implements ExternalTransformRegistrar {
+
+    @Override
+    public Map<String, Class<? extends ExternalConfigBuilder>> knownBuilders() {
+      return ImmutableMap.of(
+          "beam:external:java:generate_sequence:v1", ExternalConfiguration.class);
+    }
+
+    /** Parameters class to expose the transform to an external SDK. */
+    public static class ExternalConfiguration
+        implements ExternalConfigBuilder<PBegin, PCollection<Long>> {
```

Comment:
ExternalConfiguration extending ExternalConfigBuilder looks a bit weird and looks like we'll have to put some effort into keeping transform builder and external configuration in builder for each transform in sync. Instead how about transforms that support cross-language having to directly implement  ExternalConfigBuilder and making ExternalConfiguration POJO a parameter of the build() method ? Hopefully this will encourage developer to keep builders in sync when transforms get updated. Another option will be to use the same builder as the transform when possible.

Reply:
I'll look into that. We cannot implement `ExternalConfigBuilder` directly on the transform but ideally we would implement it on the existing (non-portable) builder.